### PR TITLE
Clean up more directory references

### PR DESCRIPTION
**Pull request by madhurajayaraman**
On-Device-Test: true

b/302730696

### CODE REVIEW

File: base/files/file_enumerator_starboard.cc [hunk 1]
```diff
@@ -112,10 +113,10 @@ FileEnumerator::~FileEnumerator() = default;
...
     const FilePath& source) {
   internal::AssertBlockingAllowed();
-  SbFileError error;
-  SbDirectory dir = SbDirectoryOpen(source.value().c_str(), &error);
-  if (!SbDirectoryIsValid(dir)) {
-    error_ = static_cast<File::Error>(error);
+
+  DIR* directory = opendir(source.value().c_str());
+  if (!(directory)) {
+    error_ = File::Error::FILE_ERROR_FAILED;
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply base/files/file_enumerator_starboard.cc:119]: Just used FAILED as a catch all, can swap it.

File: base/files/file_enumerator_unittest.cc
Comment (dahlstrom-g) [inline base/files/file_enumerator_unittest.cc:1]: The modification in this file removes code from upstream:
https://chromium.googlesource.com/chromium/src/+/114.0.5735.358/base/files/file_enumerator_unittest.cc#410

**Review by dahlstrom-g — CHANGES_REQUESTED** [review]

File: base/files/file_enumerator_unittest.cc
Comment (madhurajayaraman) [reply base/files/file_enumerator_unittest.cc:1]: Should I add an if clause for Windows here? Then we can match the file_enumerator_unittest.cc file to upstream. https://github.com/youtube/cobalt/pull/3840/files#diff-49176cb792325c5e1004e6396092a22f5dd2a9cc86cc6f4917447c3206542cf5R119

Comment (kaidokert) [reply base/files/file_enumerator_unittest.cc:1]: Yes, i think that would be correct. `BUILDFLAG(IS_WIN)` should not actually be active for our Starboard/windows based builds - that's an upstream Chromium flag and concept. 

So adding it back in _should be_ a no-op for our build configs. It's confusing and don't take my word for it ..

**Review by dahlstrom-g — APPROVED** [review]
no Chromium files modified

**Pull request merged**